server/src/clients/google.py

```diff
@@ -92,3 +92,38 @@ def get_user_info(
             parsed_user_info[field] = user_info.get(field, "")
 
         return (None, parsed_user_info)
+
+
+@error_bounded(
+    (AppError(ErrorCode.GOOGLE_API_ERROR, "Google api error - get user contacts"), None)
+)
+def get_people_connections_list(token: str,) -> Tuple[Optional[AppError], List[object]]:
+    """
+    Request list of user's contacts from Google People API.
+
+    Returns a list of Google API Person objects representing all
+    of the user's contacts
+    """
+    creds = Credentials(token) if token else None
+    with build("people", "v1", credentials=creds) as people_service:
+        pagination_finished = False
+        next_page_token = ""
+        people_list_complete = []
+        # paginate through contacts and add to list
+        while not pagination_finished:
+            people_response = (
+                people_service.people()
+                .connections()
+                .list(
+                    resourceName="people/me",
+                    personFields="names,emailAddresses",
+                    pageToken=next_page_token,
+                )
+            ).execute()
+            people_list_complete += people_response["connections"]
+            if "nextPageToken" in people_response:
+                next_page_token = people_response["nextPageToken"]
+            else:
```

Comment:
You can probably just 'break' out of the loop instead of setting the flag to true.

Reply:
mm true, do you think that's better, or is it fine to have a little extra documentation built into the naming?

(while googling python break syntax i actually got 2nd hand yelled at about using `break` by a stackOverflow thread lol: https://stackoverflow.com/questions/14594522/how-to-break-out-of-while-loop-in-python)